code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -110,7 +110,11 @@
 	parents[i] = null
 
 /obj/machinery/atmospherics/components/returnPipenetAir(datum/pipeline/reference)
-	return airs[parents.Find(reference)]
+	var/list/returned_air = list()
+	for (var/i in 1 to parents.len)
+		if (parents[i] == reference)
+			returned_air += airs[i]
```

Comment:
Early return here? For clarity and future proofing

Reply:
would this suffice?

Reply:
honestly I mean use continue if parents != reference, but it's a zero sum game. I'm not sure I like what you've done, since it's somewhat needless complexity. I'll explain better tomorrow yeah?